src/Core/Ast/Parser/NikicPhpParser/FileReferenceVisitor.php

```diff
@@ -40,16 +40,28 @@ class FileReferenceVisitor extends NodeVisitorAbstract
 
     private ReferenceBuilder $currentReference;
 
+    private readonly ?string $internalTag;
+
+    /**
+     * @param array<mixed> $config
```

Comment:
The shape of the array is known here, no? If should, it should be properly typed.

Reply:
I don't know how to represent the fact that this array should contain a specific key that maps to a string or null, and another key that maps to an array, which in turn is a list of strings.

Basically, we'd need to represent the config definition of "analyser" here. Wouldn't that mean that the anotations refering to it would break when we add another config option?

Reply:
array{key: ?string, key2: list<string>}, see https://phpstan.org/writing-php-code/phpdoc-types#array-shapes

Reply:
Ah, nice!

Is it ok to only declare the key we are actually using? Declaring all the keys that may be present in the array would be cumbersome when we want to add another one.